jetty-websocket/websocket-javax-tests/src/test/java/org/eclipse/jetty/websocket/javax/tests/RestartContextTest.java

```diff
@@ -72,12 +72,7 @@ public void testStartStopStartServletContextListener() throws Exception
         context.addEventListener(new AddEndpointListener());
 
         // Setup handler tree
-        HandlerList handlers = new HandlerList();
-        handlers.addHandler(context);
-        handlers.addHandler(new DefaultHandler());
-
-        // Add handler tree to server
-        server.setHandler(handlers);
+        server.setHandler(new HandlerList(context, new DefaultHandler()));
```

Comment:
Again, is `DefaultHandler` ever called since `ServletContextHandler` is a terminal handler?

Reply:
@sbordet `ServletContextHandler` is not always a terminal handler.  
All `ContextHandler` must first match a request before they continue handling it. They reject a match because of context path and/or virtual host and/or named connector.  Even if a request matches the context, it can still be returned unhandled if the ServletHandler is configured to have no default.